src/components/PrometheusGraph/_Legend.js

```diff
@@ -131,19 +149,27 @@ class Legend extends React.PureComponent {
         )}
         {this.state.shown && (
           <LegendItems>
-            {this.props.multiSeries.map(series => (
-              <LegendItem
-                key={series.key}
-                title={series.hoverName}
-                onClick={ev => this.handleLegendItemClick(ev, series)}
-                onMouseEnter={() => this.handleLegendItemMouseEnter(series)}
-                onMouseLeave={() => this.handleLegendItemMouseLeave()}
-                selected={this.seriesSelected(series)}
-              >
-                <ColorBox color={series.color} />
-                <LegendItemName>{series.legendName}</LegendItemName>
-              </LegendItem>
-            ))}
+            {this.props.multiSeries.map(series => {
+              const multiLine = isArray(series.legendName);
+
+              return (
+                <LegendItem
+                  key={series.key}
+                  title={series.hoverName}
+                  onClick={ev => this.handleLegendItemClick(ev, series)}
+                  onMouseEnter={() => this.handleLegendItemMouseEnter(series)}
+                  onMouseLeave={() => this.handleLegendItemMouseLeave()}
+                  selected={this.seriesSelected(series)}
+                >
+                  <ColorBox color={series.color} />
+                  <LegendItemName multiLine={multiLine}>
+                    {multiLine
+                      ? series.legendName.join('\n')
```

Comment:
I think I'd prefer to see the full joining happen in the function which generates the `legendName` to have a consistent type (string), and not having to distinguish the two cases here.

Alternatively, rename it to `legendNameParts` or something similar and make it always return an array of strings (rows), even though in most cases it will be a singleton array.